WordPressCom-Stats-iOS/WPStatsServiceRemote.m

```diff
@@ -649,8 +650,15 @@ - (AFHTTPRequestOperation *)operationForVisitsForDate:(NSDate *)date
             periodSummary.likesValue = period[likesIndex];
             periodSummary.comments = [self localizedStringForNumber:period[commentsIndex]];
             periodSummary.commentsValue = period[commentsIndex];
-            [array addObject:periodSummary];
-            dictionary[periodSummary.date] = periodSummary;
+            
+            if (periodSummary.date) {
+                [array addObject:periodSummary];
+                dictionary[periodSummary.date] = periodSummary;
+            } else {
+                DDLogError(@"operationForVisitsForDate resulted in nil date: raw date: %@", period[periodIndex]);
+                [WPAnalytics track:WPAnalyticsStatLogSpecialCondition withProperties:@{@"error_condition" : @"WPStatsServiceRemote operationForVisitsForDate:andUnit:withCompletionHandler",
+                                                                                       @"error_details" : [NSString stringWithFormat:@"Date in raw format: %@", period[periodIndex]] }];
```

Comment:
What if the period is NSNotFound?. I know that shouldn't ever happen... but what if?